src/librbd/internal.cc

```diff
@@ -1351,9 +1351,7 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
     int r = ictx->state->open(true);
     if (r < 0) {
       ldout(cct, 2) << "error opening image: " << cpp_strerror(-r) << dendl;
-      if (r != -ENOENT) {
```

Comment:
I believe the intent of this was the handle the case where you had a crash between adding the image to the trash and removing the image's id object and removing the image from the directory (i.e. run it a second time to get to a known good state). However, since the image id won't be populated, the current implementation is not correct. I think on -ENOENT, you need to do an additional check to retrieve the image id from the directory.